mkosi/machine.py

```diff
@@ -0,0 +1,133 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from __future__ import annotations
+
+import contextlib
+import signal
+import subprocess
+from textwrap import dedent
+from typing import Any, Optional, Sequence
+
+import pexpect  # type: ignore
+
+from . import (
+    MKOSI_COMMANDS_SUDO,
+    CompletedProcess,
+    build_stuff,
+    check_native,
+    check_output,
+    check_root,
+    init_namespace,
+    load_args,
+    needs_build,
+    parse_args,
+    prepend_to_environ_path,
+    run_command_image,
+    run_qemu_cmdline,
+    run_shell_cmdline,
+    unlink_output,
+)
+from .backend import MkosiArgs, die
+
+
+class Machine:
+    def __init__(self, args: Optional[Sequence[str]] = None, debug: bool = False) -> None:
+        # Remains None until image is built and booted, then receives pexpect process
+        self._serial: Optional[pexpect.spawn] = None
+        self.exit_code: int = -1
+        self.debug = debug
+        self.stack = contextlib.ExitStack()
+        self.args: MkosiArgs
+
+        # We make sure to add the arguments in the machine class itself, rather than typing this for every testing function.
+        tmp = parse_args(args)["default"]
+        tmp.force = 1
+        tmp.autologin = True
+        if tmp.verb == "qemu":
+            tmp.bootable = True
+            tmp.qemu_headless = True
+            tmp.hostonly_initrd = True
+            tmp.network_veth = True
+            tmp.ssh = True
+        elif tmp.verb == "boot":
+            pass
+        else:
+            die("No valid verb was entered.")
+
+        self.args = load_args(tmp)
+
+
```

Comment:
Need to be consistent with the whitespace between class methods as well. 2 newlines between functions in a file. 1 newline between methods in a class. I know its weird but it's how we do it everywhere so we should stick to it.

Reply:
Oh, I really thought it was 2, since functions are the ones appearing the most. Will write that down so I don't forget.

Reply:
This is just [PEP 8](https://www.python.org/dev/peps/pep-0008/#blank-lines) :)